File: 1796 second-largest-digit-in-a-string/solution.java
class Solution {
    public int secondHighest(String s) {
        int fmax = -1, smax = -1;
        for (char i : s.toCharArray()) {
            if (Character.isDigit(i)) {
                int num = (int) (i) - 48;
                if (num > fmax) {
                    smax = fmax;
                    fmax = num;
                } else if (num < fmax && num > smax) {
                    smax = num;
                }
            }

        }
        return smax;
    }
}

File: 1796 second-largest-digit-in-a-string/solution.cs
public class Solution {
    public int SecondHighest(string s) {
        int fmax=-1,smax=-1;
        foreach(char i in s) {
            if(Char.IsDigit(i)) {
                int num=(int)i-48;
                if(num>fmax){
                    smax=fmax;
                    fmax=num; 
                } else if(num>smax && num<fmax) {
                    smax=num;
                }
            }
        }
        return smax;
    }
}
